app/assets/javascripts/admin/products/directives/set_variant_on_demand.js.coffee

```diff
@@ -0,0 +1,21 @@
+angular.module("admin.products").directive "setOnDemand", ->
+  link: (scope, element, attr) ->
+    onHand = element.context.querySelector("#variant_on_hand")
+    onDemand = element.context.querySelector("#variant_on_demand")
+
+    if onDemand.checked
+      onHand.disabled = 'disabled'
+      onHand.dataStock = onHand.value
+      onHand.value = t('admin.products.variants.infinity')
+
+    onDemand.addEventListener 'change', (event) ->
+      if onDemand.checked
+        onHand.disabled = 'disabled'
+        onHand.dataStock = onHand.value
+        onHand.value = t('admin.products.variants.infinity')
```

Comment:
How can I refactor these 3 lines to be on a function ? They are called twice, it's not ideal. I didn't manage to declare a function.

Reply:
I just looked up some other directives in our codebase, I think you can use a named function:


```ruby
angular.module("admin.products").directive "setOnDemand", ->
  link: (scope, element, attr) ->
    onHand = element.context.querySelector("#variant_on_hand")
    onDemand = element.context.querySelector("#variant_on_demand")

     disableOnHandIfOnDemand()

     onDemand.addEventListener 'change', (event) ->
         disableOnHandIfOnDemand()

     disableOnHandIfOnDemand = ->
         if onDemand.checked        
            onHand.disabled = 'disabled'
            onHand.dataStock = onHand.value
            onHand.value = t('admin.products.variants.infinity')
```

Reply:
Nice thanks!